### PR TITLE
GLTF Draco Integration Improvements

### PR DESCRIPTION
**Pull request by ibgreen**
Ready for review!

### CODE REVIEW

File: lerna.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "lerna": "2.9.1",
-  "version": "0.7.2",
+  "version": "0.8.0",
```
Comment (georgios-uber) [inline lerna.json]: This will be done automatically when you publish

File: modules/gltf/src/glb/gltf-types.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+// TODO - Remove: glTF constants should not be know to the GLB layer
```
Comment (georgios-uber) [inline modules/gltf/src/glb/gltf-types.js]: I think we should move this to luma.gl

Please remove if not referenced here.

Comment (ibgreen) [reply modules/gltf/src/glb/gltf-types.js]: > I think we should move this to luma.gl. Please remove if not referenced here.

Problem is they are referenced here. I have a bullet in the GLTF Tracker to refactor this.

File: modules/gltf/src/gltf/gltf-parser.js [hunk 3]
```diff
@@ -186,305 +201,121 @@ export default class GLTFParser {
...
+        primitive.indices = decodedData.indices;
+      } else {
+        delete primitive.indices;
       }
     } finally {
       dracoDecoder.destroy();
     }
 
-    return unpackedPrimitive;
+    // Extension has been processed, delete it
```
Comment (georgios-uber) [inline modules/gltf/src/gltf/gltf-parser.js]: ?

File: lerna.json (hunk 1 above)
Comment (ibgreen) [reply lerna.json]: I guess our publish script would need to pass a parameter I am not quite familiar with how that works.

File: modules/gltf/src/gltf/gltf-parser.js [hunk 4]
```diff
@@ -186,305 +201,121 @@ export default class GLTFParser {
...
 
-    const dracoDecoder = new this.DracoDecoder();
+    const dracoDecoder = new options.DracoDecoder();
 
     try {
-      const decodedData = dracoDecoder.decodeMesh(compressedMesh);
-      unpackedPrimitive.attributes = decodedData.attributes;
+      const buffer = this._getBufferViewArray(compressedMesh.bufferView);
+      const decodedData = dracoDecoder.decodeMesh(buffer);
+      primitive.attributes = decodedData.attributes;
```
Comment (georgios-uber) [inline modules/gltf/src/gltf/gltf-parser.js]: Great, attach to same object.

**Pull request merged**